datafusion/common/src/file_options/file_type.rs

```diff
@@ -41,7 +43,8 @@ pub trait GetExt {
 }
 
 /// Readable file type
-#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+#[allow(clippy::derived_hash_with_manual_eq)]
```

Comment:
I think the manual and auto traits should be compatible, so silencing this is OK.